### ISSUE TITLE
Expiration time in variable

### ISSUE THREAD

**Issue by katgirl** [contao/contao#5770]
### Affected version(s)

4.13.15

### Description

The time should not be constant in it, but better can be set via a variable to adjust the time.

See also #5769

https://github.com/contao/contao/blob/57ea584579919a096ec0ff5b43246c5bbfa3b98f/core-bundle/contao/models/MemberModel.php#L247

**Comment by fritzmg** [comment]
This would be a new feature, not a bug, no?

**Comment by katgirl** [comment]
Can you change the labels?

**Comment by leofeyer** [comment]
AFAIK, it is required to delete the data after 24 hours, therefore I don‘t see a point in making this configurable?

**Comment by fritzmg** [comment]
That depends on the law your site is subjected to, doesn't it? It's probably not the same in every country.

**Comment by leofeyer** [comment]
IDK. At least in all GDPR-regulated countries, it should be the same.

**Comment by katgirl** [comment]
We use the registration and only then manually check the authorization. If the registration is done on Friday, the customer service can already not do the check on Monday. Therefore we need the possibility to adjust this.

**Comment by fritzmg** [comment]
@katgirl if the registrations are manually checked anyway, why do you use the registration activation?

**Comment by katgirl** [comment]
The activation link goes to customer support, who can then send the activation email over the link. We use the notificationcenter.

We wanted to develop as little as possible in the project itself and otherwise we have seen no way that the customer support (without backend and own email to the customer) can notify the customer.

**Comment by fritzmg** [comment]
In this case I would recommended to implement a custom token for the notification center.

The confirm link sent by Contao is not really intended to be used that way.

**Comment by aschempp** [comment]
The token is not related to the notification center, but to activate the user. And re-implementing user activation just to make this use case possible seems overkill. I'd fully agree that this should be configurable, especially because someone NOT under GDPR anywhere else in the world might set this to a year or whatever.

**Comment by fritzmg** [comment]
I agree that it should be configurable, but I disagree that the existing functionality should be used for such a use case.

**Comment by d0p3fish** [comment]
I have the same problem. New users register during regular working hours, such as on Friday, and then want to activate their account on Monday. However, the token has expired.

**Comment by leofeyer** [comment]
If a token is not confirmed within 24 hours, the GDPR requires to delete it. There is nothing we can do about it.

**Comment by aschempp** [comment]
> If a token is not confirmed within 24 hours, the GDPR requires to delete it. There is nothing we can do about it.

what we _can_ do, is allow people to ignore the law or adjust it to something that is not GDPR (e.g. because they are not subjected to the law). Assuming that GDPR applies to everyone is wrong imho, and I can probably ignore the law e.g. in a B2B setting (intranet etc.).

**Comment by leofeyer** [comment]
Sure, feel free to create a pull request.

**Comment by aschempp** [comment]
> Sure, feel free to create a pull request.

👍  that would be up to @katgirl though, as she needs that feature 😅

**Comment by d0p3fish** [comment]
> If a token is not confirmed within 24 hours, the GDPR requires to delete it. There is nothing we can do about it.

Which law is this exactly? I can't find anything about deleting users within 24 hours. Also, ChatGPT says it's not required.

**Comment by fritzmg** [comment]
As far as I can see the GDPR does not determine a fixed time frame for such cases. It only says that you must not keep data longer than necessary. Keeping unconfirmed accounts/email addresses for 24 hours is a sensible default for the intended use case.

**Comment by kreadiv** [comment]
A client of mine has the same problem. From my point of view the token should be configurable.

**Comment by zoglo** [comment]
> > Sure, feel free to create a pull request.
> 
> 👍 that would be up to katgirl though, as she needs that feature 😅

As already mentioned, this feature was requested but nobody did a PR yet.

@kreadiv Maybe you wanna do a PR for that? However, this would go into 5.6 as the deadline for 5.5 did end 2 weeks ago.

**Comment by kreadiv** [comment]
@zoglo honestly iam not that deep in the core. But if i find the time i will gladly make a PR.
